### ISSUE TITLE
Can't disable source maps for create-react-app

### ISSUE THREAD

**Issue by johannescpk** [cypress-io/cypress#9701]
Thanks for working on cypress component testing!

Currently when using the `reactScripts` plugin there's no way to disable the source map generation from webpack, since `find-webpack` hardcodes `NODE_ENV` to `development` and CRA only allows disabling source maps with `GENERATE_SOURCEMAP` in production builds.

What would be the preferred way to allow doing this? Passing some sort of webpack config overwrite/merge function to `reactScripts` as argument? Some sort of ENV variable?

My workaround currently is to hardcode

```js
preprocessorOptions.webpackOptions.devtool = false
```

at the end of `cra-v3/file-preprocessor.js`

**Comment by lmiller1990** [comment]
Hi @johannescpk,

We just released Cypress 7.0, which should give you more flexibility with configuring Cypress for component testing.

7.0 comes with a completely revamped Component Testing experience. Component testing is no longer experimental (but still in alpha, with lots of updates and improvements coming in the next few weeks). Please let us know if you find any more issues.

Updated Component Testing intro: https://docs.cypress.io/guides/component-testing/introduction
Changelog: https://docs.cypress.io/guides/references/changelog
Migration guide: https://docs.cypress.io/guides/references/migration-guide#Component-Testing

The main difference is we moved from a `file:preprocessor` architecture to a dev-server architecture. Component testing is no longer experimental, and has a dedicated runner (which is build on top of an extends the existing cypress runner).

The main benefits of the improved component testing runner and architecture are:

- things are now (much) faster
- no need to launch into the electron shell. It opens straight to the browser (select your spec from the inline spec selector)

Regarding your specific request, the way we recommend most people to inject the dev-server (previously file:preprocessor) is like this:

```js
// cypress/plugins/index.js

const injectDevServer = require("@cypress/react/plugins/react-scripts")

module.exports = (on, config) => {
  injectDevServer(on, config)
  return config
}
```

If you want more flexibility (like you do), you can see how [the plugin works](https://github.com/cypress-io/cypress/blob/develop/npm/react/plugins/react-scripts/index.js)

```js
const { startDevServer } = require('@cypress/webpack-dev-server')
const findReactScriptsWebpackConfig = require('./findReactScriptsWebpackConfig')

module.exports = (on, config) => {
  on('dev-server:start', async (options) => {
    return startDevServer({ 
      options, 
      webpackConfig: findReactScriptsWebpackConfig(config) 
    })
  })

  // IMPORTANT to return the config object
  // with the any changed environment variables
  return config
}
```

So what you should be able to do is:

1. find your react-scripts webpack config (using `find-webpack` or [this function](https://github.com/cypress-io/cypress/blob/develop/npm/react/plugins/react-scripts/findReactScriptsWebpackConfig.js))
2. in `cypress/plugins/index.js`, when you do the `dev-server:start` event and pass in your `webpackConfig`, modify the webpack config there
3. This should allow you to do whatever modifications you see fit.

You will need Cypess 7.x and the new dev-server architecture and dedicated component testing runner (which you open with `yarn cypress open-ct`, [see the migration docs for more information](https://docs.cypress.io/guides/references/migration-guide#Component-Testing).

Lot of information, let me know if you need help with updating, or if you run into any problems.

**Comment by jennifer-shehane** [comment]
Closing as resolved. 

If you're experiencing a bug similar to this in Cypress, please [open a new issue](https://github.com/cypress-io/cypress/issues/new/choose) with a fully reproducible example that we can run. There may be a specific edge case with the issue that we need more detail to fix.
